contrib/linux_build_toolchain.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+# Script to mimic how CI is building individual toolchain.
+
+# Default is to use current directory as the working directory.
+# SDK is under ${WORKING_ROOT_DIR}/sdk-ng/
+# Build directory is ${WORKING_ROOT_DIR}/build-workspace/
+WORKING_ROOT_DIR=${PWD}
+SDK_DIR=${WORKING_ROOT_DIR}/sdk-ng
+BUILD_DIR=${WORKING_ROOT_DIR}/build-workspace
+
+while getopts "hc:s:o:" ARGS; do
+	case $ARGS in
+	h)
+		echo "Usage: $0 -c config file name [-s SDK directory] [-o Build directory] [-h]"
+		echo ""
+		echo "Arguments:"
+		echo ""
+		echo "  -c Name of config file (without extension .config)"
+		echo ""
+		echo "  -s SDK directory. Optional."
+		echo "     Default is ${GITHUB_WORKSPACE}"
+		echo ""
+		echo "  -o Build directory, for building artifacts and final output. Optional."
+		echo "     Default is ${WORKSPACE}"
+		echo ""
+		echo "  -h This help."
+		echo ""
+		exit 1
+		;;
+	c)
+		CONFIG_FILE_NAME=$OPTARG
+		;;
+	s)
+		SDK_DIR=$OPTARG
+		;;
+	o)
+		BUILD=$OPTARG
+		;;
+	*)
+		echo "Unknown arguments: $1"
+		exit 1
+		;;
+	esac
+done
+
+# These variables are the same as the CI script.
+export GITHUB_WORKSPACE=${SDK_DIR}
+export WORKSPACE=${BUILD_DIR}
+
+export CT_PREFIX=${WORKSPACE}/output
+export CT_NG=${WORKSPACE}/crosstool-ng/bin/ct-ng
+
+TOOLCHAIN_OUTPUT_DIR=${WORKSPACE}/output/${CONFIG_FILE_NAME}
+
+echo "SDK directory: ${GITHUB_WORKSPACE}"
+echo "Build directory: ${WORKSPACE}"
+
+SDK_CONFIG_FILE="${GITHUB_WORKSPACE}/configs/${CONFIG_FILE_NAME}.config"
+
+if [ ! -f "${SDK_CONFIG_FILE}" ]; then
+	echo "ERROR: ${SDK_CONFIG_FILE} does not exist!"
+	exit 1
+fi
+
+echo "Build config file: ${SDK_CONFIG_FILE}"
+echo "Toolchain output directory: ${TOOLCHAIN_OUTPUT_DIR}"
+
+if [ ! -x "${WORKSPACE}/crosstool-ng/bin/ct-ng" ]; then
+	echo "========== Building ct-ng =========="
+
+	# Create build directory
+	pushd ${WORKSPACE}
+	mkdir -p crosstool-ng-build
+	cd crosstool-ng-build
+
+	# Bootstrap crosstool-ng
+	pushd ${GITHUB_WORKSPACE}/crosstool-ng
+	./bootstrap
+	popd
+
+	# Build and install crosstool-ng
+	${GITHUB_WORKSPACE}/crosstool-ng/configure --prefix=${WORKSPACE}/crosstool-ng
+	make
+	make install
+
+	# Clean up build directory to reduce disk usage
+	popd
+	rm -rf ${WORKSPACE}/crosstool-ng-build
```

Comment:
It would be better to leave the build directory as is for a local build for debugging and incremental build purposes.

Reply:
removed.